src/scss/components/modalMediaViewer.scss

```diff
@@ -0,0 +1,185 @@
+@use "sass:map";
+@use "../global.scss";
+
+// Modal Media Viewer Styles
+#modalMediaViewer {
+  .modal-dialog {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: 100vh;
+    margin: 0;
+    max-width: 100%;
+  }
+
+  .modal-content {
+    background-color: rgba(0, 0, 0, 0.9);
+    border: none;
+    height: 100vh;
+    width: 100vw;
+    overflow: hidden;
+  }
+
+  .modal-body {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 0;
+    height: 100%;
+    width: 100%;
+    overflow: hidden;
+    position: relative; // Important for control positioning
+  }
+
+  // Reset mediaDiv to original size
+  #mediaDiv {
+    width: 90vw;
+    height: 90vh;
+    overflow: hidden;
+    position: relative;
+
+    // Make mediaDiv a flex container to properly contain carousel
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  // Carousel fills available space and uses flexbox
+  .carousel {
+    width: 100%;
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+    position: relative;
+  }
+
+  // Carousel inner grows to use available space
+  .carousel-inner {
+    flex: 1;
+    min-height: 0; // Allow shrinking
+    position: relative;
+    overflow: hidden;
+  }
+
+  // Style active carousel items only
+  .carousel-item.active {
+    height: 100%;
+    display: flex !important;
+    align-items: center;
+    justify-content: center;
+  }
+
+  // Remove padding on any element (for aspect ratio padding)
+  #mediaDiv [style*="padding-top"] {
+    padding-top: 0 !important;
+  }
+
+  // Constrain content within the modal
+  .carousel-item > * {
+    max-width: 100%;
+    max-height: 100%;
+  }
+
+  // All media should be contained
+  img, video, iframe,
```

Comment:
can we do this with a class and domfactory? it might clobber other img video iframe, etc on other pages